src/network/networkd-address.c

```diff
@@ -918,6 +1011,56 @@ int config_parse_address_flags(const char *unit,
         return 0;
 }
 
+int config_parse_duplicate_address_detection(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        Network *network = userdata;
+        _cleanup_(address_free_or_set_invalidp) Address *n = NULL;
+        AddressFamily a;
+        int r;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = address_new_static(network, filename, section_line, &n);
+        if (r < 0)
+                return r;
+
+        r = parse_boolean(rvalue);
+        if (r >= 0) {
+                if (r)
+                        log_syntax(unit, LOG_WARNING, filename, line, 0,
+                                   "For historical reason, %s=%s means %s=no. "
+                                   "If you want to enable %s=, use 'ipv4', 'ipv6' or 'both'.",
+                                   lvalue, rvalue, lvalue, lvalue);
+                n->duplicate_address_detection = ADDRESS_FAMILY_NO;
```

Comment:
Do I understand correctly:
Previously: `DuplicateAddressDetection=yes` → `address->flags |= IFA_F_NODAD`;
                  `DuplicateAddressDetection=no` → the flag is not set, so the kernel does DAD.
But now, both settings map to `n->duplicate_address_detection = ADDRESS_FAMILY_NO`.
Shouldn't this be
```C
 n->duplicate_address_detection = r ? ADDRESS_FAMILY_NO : ADDRESS_FAMILY_YES;
```
?

Reply:
Your understanding is right. However, if your proposal is adapted, then we have no way to disable DAD completely... Previously, the default was false, no NODAD flag. So, I hope no one set to false.

Reply:
Or, how about your proposal + 'none', which is mapped to ADDRESS_FAMILY_NO?

Reply:
Yes, please!